lib/simulator-management.js

```diff
@@ -40,17 +39,18 @@ async function runSimulatorReset (device, opts) {
     return;
   }
 
-  // The simulator process must be ended before we delete applications.
-  await endSimulator(device);
-
   if (opts.fullReset) {
-    // noReset === false && fullReset === true
     log.debug('Reset: fullReset is on. Cleaning simulator');
+    // The simulator process must be ended before we delete applications.
+    await endSimulator(device);
     await fullResetSimulator(device);
-  } else {
-    // noReset === false && fullReset === false
-    log.debug(`Reset: fullReset is not on. Performing 'fast reset' by removing app resources`);
-    await resetSimulator(device, opts);
+  } else if (opts.bundleId) {
+    // Terminate the app under test if it is still running on Simulator
+    try {
+      await terminate(device.udid, opts.bundleId);
```

Comment:
Probably, we'll need to do it only in case opts.app === null, because uninstall already does the thing for third-party apps. But I agree, it is necessary to put it back.